libraries/stdlib/src/kotlin/io/Console.kt

```diff
@@ -171,11 +171,21 @@ private val stdin: BufferedReader by lazy { BufferedReader(InputStreamReader(obj
     public override fun read(b: ByteArray, off: Int, len: Int): Int {
         return System.`in`.read(b, off, len)
     }
-}))}
+})}
 
 /**
  * Reads a line of input from the standard input stream.
  *
  * @return the line read or `null` if the input stream is redirected to a file and the end of file has been reached.
  */
-public fun readLine(): String? = stdin.readLine()
+public fun readLine(): String? {
+    tailrec fun readChars(oldBuffer: StringBuffer? = null): StringBuffer? {
+        val c = mystdin.read()
+        if(c < 0) return oldBuffer
+        val buffer = oldBuffer ?: StringBuffer()
+        val ch = c.toChar()
+        if(ch == '\r' || ch == '\n') return buffer
+        return readChars(buffer.append(ch))
+    }
+    return readChars()?.toString()
+}
```

Comment:
Overall, `buildString` + imperative loop might be more clear than tail recursion approach.